src/rgw/rgw_file.h

```diff
@@ -802,7 +802,10 @@ namespace rgw {
 			     const uint32_t flags = RGWFileHandle::FLAG_NONE) {
       using std::get;
 
-      LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      // This seems like a Clang compiler error.
+      // It result in irgw::RGWFileHandle::FLAG_NONE being undefined during linking.
+      // LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      LookupFHResult fhr { nullptr, 0 };
```

Comment:
@wjwithagen i tried with a simple example like
```c++
/// hello.cc
#include <cstdint>

struct Foo
{
  static constexpr uint32_t FOO = 0;
};

struct Bar
{
  int foo(const uint32_t flags = Foo::FOO) {
    return 0;
  }
};
```

```c++
// world.cc
#include "hello.cc"

int main()
{
  Bar b;
  b.foo();
}
```

```makefile
# Makefile
all:
    clang++-3.7 -std=c++11 -O0 -c hello.cc
    clang++-3.7 -std=c++11 -O0 -g world.cc -lm hello.o -o world
```

it compiles fine. could you show me a reproducer?